### PR TITLE
WIP Jealous

### PR DESCRIPTION
**Pull request by Eretzu**

### CODE REVIEW

File: server/src/game/cards/10-experimental/Jealous.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+	private onTargetSelected(target: ServerCard): void {
+		this.game.players.forEach(player => {
+			toDiscardHand = []
+			player.cardHand.unitCards.forEach(card => {
+				if (card.class === target.class) toDiscardHand.push(card)
+			});
+			toDiscardHand.forEach(card => {
+				player.cardHand.discardCard(card)
+			});
```
Comment (Tenebrie) [inline server/src/game/cards/10-experimental/Jealous.ts:43]:
```suggestion
player.cardHand.unitCards.filter(card => card.class === target.class).forEach(card => player.cardHand.discardCard(card))
```
Wouldn't that be the same?

Comment (Eretzu) [reply server/src/game/cards/10-experimental/Jealous.ts:43]: not exactly sure on the safety aspects
In some language messing with array while inside for loop is not cool, so I kept it safe
Will fix

Comment (Tenebrie) [reply server/src/game/cards/10-experimental/Jealous.ts:43]: Filter creates a copy, so it's perfectly fine in any case

Comment (Eretzu) [reply server/src/game/cards/10-experimental/Jealous.ts]: fixed

File: server/src/game/cards/10-experimental/HeroJealous.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+			features: [CardFeature.KEYWORD_DEPLOY],
+			stats: {
+				power: 1,
+			},
+			expansionSet: ExpansionSet.BASE,
+			isExperimental: true,
+		})
+
+		this.createDeployEffectTargets()
+			.totalTargets(1)
```
Comment (Tenebrie) [inline server/src/game/cards/10-experimental/HeroJealous.ts:29]: You can just remove the `.totalTargets(1)`. It's the default value.

File: server/src/game/cards/10-experimental/HeroJealous.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		this.createEffect(GameEventType.CARD_TARGET_SELECTED_CARD)
+			.perform(({ targetCard }) => this.onTargetSelected(targetCard))
+			
+		// TODO On death: remove from game
+	}
+
+	private onTargetSelected(target: ServerCard): void {
+		let affected: ServerCard[] = []
+		this.game.players.forEach((player) => affected.concat(discardCopiesFromHand(target, player)))
+		this.game.animation.play(ServerAnimation.cardAffectsCards(this, affected))
```
Comment (Tenebrie) [inline server/src/game/cards/10-experimental/HeroJealous.ts:42]: You are playing animation for cards that are no longer in play. Or at least they are now in graveyard. As both the deck and the graveyard don't have graphical representations of cards, there will be no animation at all. You should be playing it before discarding the card.

Also you can start animation thread for every discard and play an animation for that one card inside the thread.

File: client/src/Pixi/locales/en.json [hunk 4]
```diff
@@ -757,5 +757,8 @@
 	"card.unitUrbanAmbusher.description": "Has +{bonusPower} Power for every Building on the same row.",
 
 	"card.unitUrbanTactician.name": "Urban Tactician",
-	"card.unitUrbanTactician.description": "Your other units that are directly adjacent to a Building have +{bonusPower} Power.<p><i>Doesn't work on buildings.</i>"
+	"card.unitUrbanTactician.description": "Your other units that are directly adjacent to a Building have +{bonusPower} Power.<p><i>Doesn't work on buildings.</i>",
+	
+	"card.heroJealous.name": "Jealous",
+	"card.heroJealous.description": "*Deploy:*\nName a unit, both players discard all copies of those units and draw as many replacements.<p>When Jealous would die it is removed from the game instead."
```
Comment (Tenebrie) [inline client/src/Pixi/locales/en.json:763]: Legendary units should also have a title. Currently your name feels more like a title (I'd just change it to something like The Jealous One).

File: server/src/game/cards/10-experimental/HeroJealous.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import ServerCard from '@src/game/models/ServerCard'
+import ServerGame from '@src/game/models/ServerGame'
+
+export default class HeroJealous extends ServerCard {
+	constructor(game: ServerGame) {
+		super(game, {
+			type: CardType.UNIT,
+			color: CardColor.GOLDEN,
+			faction: CardFaction.NEUTRAL,
+			features: [CardFeature.KEYWORD_DEPLOY],
```
Comment (Tenebrie) [inline server/src/game/cards/10-experimental/HeroJealous.ts:20]: `KEYWORD_DISCARD` is also needed

**Pull request closed without merging**